providers/provider_data_test.go

```diff
@@ -408,11 +417,42 @@ func TestProviderData_buildSessionFromClaims(t *testing.T) {
 				PreferredUsername: "Jane Dobbs",
 			},
 		},
+		"Request claim from ProfileURL": {
+			IDToken:                   minimalIDToken,
+			SetProfileURL:             true,
+			ExpectedProfileURLRequest: &profileURLRequest{Response: "{}"},
+			ExpectedSession:           &sessions.SessionState{},
+		},
+		"Skip claims request to ProfileURL": {
+			IDToken:                  minimalIDToken,
+			SetProfileURL:            true,
+			SkipClaimsFromProfileURL: true,
+			ExpectedSession:          &sessions.SessionState{},
+		},
 	}
 	for testName, tc := range testCases {
 		t.Run(testName, func(t *testing.T) {
 			g := NewWithT(t)
 
+			var (
+				profileURL        *url.URL
+				profileURLTracker mock.Mock
+			)
+			if tc.SetProfileURL {
+				profileURLSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
I'm fine with setting up a http test server here. We do something quite similar in `claim_extractor_test.go` for the profileURL as well.

Reply:
But do we really need to introduce a new go dependency for this "simple" test?

Couldn't we just introduce a simple custom handler like in `claim_extractor_test.go#L556`:

https://github.com/oauth2-proxy/oauth2-proxy/blob/52ad31752d0d0b38673d419e56ddfc4cb2f56857/pkg/providers/util/claim_extractor_test.go#L556-L564


This way we don't need to introduce a new dependency to objx and mock.Mock and we will not need to the assertion in line 485

Reply:
Agree. The mock.Mock package is completely overkill here. Only need to test that a request to profileURL is sent or not.
Fixed in ceccfb2141ddc63f9728097a4c4ae1e4f8cefd9a